connector_jira/models/account_analytic_line/common.py

```diff
@@ -171,6 +180,68 @@ def _compute_jira_references(self):
             record.jira_epic_issue_key = main_binding.jira_epic_issue_key
             record.jira_epic_issue_url = main_binding.jira_epic_issue_url
 
+    @api.model
+    def _get_connector_jira_fields(self):
+        return [
+            'jira_bind_ids',
+            'project_id',
+            'task_id',
+            'user_id',
+            'employee_id',
+            'name',
+            'date',
+            'unit_amount',
+        ]
+
+    @api.model
+    def _connector_jira_create_validate(self, vals):
+        ProjectProject = self.env['project.project']
+        project_id = vals.get('project_id')
+        if project_id:
+            project_id = ProjectProject.sudo().browse(project_id)
+            if not self.env.context.get('connector_jira') and \
+                    project_id.mapped('jira_bind_ids')._is_linked():
+                raise exceptions.UserError(_(
+                    'Timesheet can not be created in project linked to JIRA!'
+                ))
+
+    @api.multi
+    def _connector_jira_write_validate(self, vals):
+        if not self.env.context.get('connector_jira') and \
+                any(f in vals
+                    and normalize_field_value(self._fields[f], vals[f]) != x[f]
```

Comment:
not easy to read IMO

Reply:
Do you think it would worth extracting that?

Reply:
just split it in blocks (eg: if + nested for loops)

Reply:
Done